components/style/stylist.rs

```diff
@@ -470,17 +463,49 @@ impl Stylist {
                     for selector in &style_rule.selectors.0 {
                         self.num_selectors += 1;
 
+                        let map = match selector.pseudo_element() {
+                            None => &mut origin_cascade_data.element_map,
+                            Some(pseudo) => {
+                                if pseudo.is_precomputed() {
+                                    if !selector.is_universal() ||
+                                       !matches!(origin, Origin::UserAgent) {
+                                        // ::-moz-tree selectors may appear in
+                                        // non-UA sheets (even though they never
+                                        // match).
+                                        continue;
+                                    }
+
+                                    self.precomputed_pseudo_element_decls
```

Comment:
(It would be nice if we could handle the precomputed case separately, above the match, so we can can all of the normal rule collection bits together, but it's not a big deal.)